sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -479,7 +479,7 @@ case class Cast(child: Expression, dataType: DataType) extends UnaryExpression w
     case DoubleType => castToDoubleCode(from)
 
     case array: ArrayType =>
-      castArrayCode(from.asInstanceOf[ArrayType].elementType, array.elementType, ctx)
+      castArrayCode(from.asInstanceOf[ArrayType], array, ctx)
```

Comment:
how about we move this into another PR? I think the main purpose of this PR is to eliminate the unnecessary Cast